src/bpftrace.h

```diff
@@ -182,7 +184,7 @@ class BPFtrace
   int print_map_stats(IMap &map);
   int print_hist(const std::vector<uint64_t> &values, uint32_t div) const;
   int print_lhist(const std::vector<uint64_t> &values, int min, int max, int step) const;
-  static uint64_t reduce_value(const std::vector<uint8_t> &value, int ncpus);
+  template <typename T> static T reduce_value(const std::vector<uint8_t> &value, int ncpus);
   static int64_t min_value(const std::vector<uint8_t> &value, int ncpus);
   static uint64_t max_value(const std::vector<uint8_t> &value, int ncpus);
```

Comment:
I initially did it and then strippd it out:

- Min does some offsetting giving it only 32 bits of range and it already prints negative numbers correctly.
- Max does not work as `CreateMapLookupElem` returns 0 when no element exists so it can never go negative anyway.

Just templating the printers without fixing those issues seemed a bit pointless. Besides that max cannot be tested without fixing it.

I think it is better to fix them in a follow up PR than to make this one even bigger. I already have a WIP for max which I plan open once this and int casts are merged and I hope to do min after that.